src/realm/object-store/shared_realm.cpp

```diff
@@ -789,6 +789,12 @@ void Realm::run_writes()
         // writes as we can't add commits while in that state
         return;
     }
+    if (is_in_transaction()) {
```

Comment:
I guess this check can cause starvation in case we have a lot of sync writes competing with async writes. I don't know if this is a realm problem, but maybe if we notice issues around this, we might think about adding some heuristic.